### PR TITLE
Add test to check all stages completed

### PR DESCRIPTION
**Pull request by CathalOConnorRH**
add test to verify that all stages in the integreatly-operator CR report completed

# Description

Added a test to check that all RHMI stages are in a completed status. Product status is tested in another test. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: test/common/integreatly_stages.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import (
+	"fmt"
+	"testing"
+
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+)
+
+var (
+	products = []v1alpha1.ProductName{
+
```
Comment (JameelB) [inline test/common/integreatly_stages.go:12]: @CathalOConnorRH Do we also need to check for bootstrap here?

**Comment by JameelB** [comment]
@CathalOConnorRH Can you also create a PR to update the test case itself with further details on what actually is being tested?

File: test/common/integreatly_stages.go (hunk 1 above)
Comment (CathalOConnorRH) [reply test/common/integreatly_stages.go:12]: Good spot. That actually isn't needed anymore. That was left over from a previous attempt at writing this. This test only tests the stages and not the product status. That's covered in another test.

File: test/common/integreatly_stages.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	if err != nil {
+		t.Fatalf("error getting RHMI CR: %v", err)
+	}
+
+	//iterate stages and check their status
+	// This does not check the pro
+	for _, stage := range rhmi.Status.Stages {
+		err := checkStatus(stage)
+		if err != nil {
+			t.Logf("Error: Product %v not completed. It's current status is %v", stage.Name, err)
```
Comment (JameelB) [inline test/common/integreatly_stages.go:23]: Should this be `t.Fatalf` instead of a log?

Comment (CathalOConnorRH) [reply test/common/integreatly_stages.go]: yeah. it should. Will push an update

**Comment by CathalOConnorRH** [comment]
Closing this as it's covered in more detail in https://issues.redhat.com/browse/INTLY-7424

**Pull request closed without merging**